coalib/results/Diff.py

```diff
@@ -70,6 +71,23 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unidiff(cls, diff_array):
+        """
+        Creates a Diff object from given unified diff.
+        :param diff_array: list of lines of unified diff.
+        """
+        source_array = []
+        target_array = []
+        patch_set = unidiff.PatchSet(diff_array)
+        for patched_file in patch_set:
+            for hunk in patched_file:
+                for line in hunk.source_lines():
+                    source_array.append(str(line)[1:])
```

Comment:
hm but @meetmangukiya is right, I don't think the coala core is able to handle that / does not propose valid patches then for a file when we use that for linter (coala uses `Diff.modified`)^^ Maybe you should take in the original file as an argument?